tests/test_helpers.py

```diff
@@ -477,3 +479,11 @@ def build_custom_services(
 def get_first_service_by_type(asset, service_type: ServiceType) -> Service:
     """Return the first Service with the given ServiceType."""
     return next((service for service in asset.services if service.type == service_type))
+
+
+def get_free_c2d_env():
+    environments = get_c2d_environments()
+    for env in environments:
```

Comment:
I would sugggest a comprehension, i.e:
`free_envs = [env for env in get_c2d_environments() if float(env["priceMin"]) == 0`
`return free_envs[0] if free_envs else None`

please check the conversion as well, I don't think the priceMin will be an int/float, but probably a string, so we need to convert it.